src/ngraph/runtime/gpu/cuda_emitter.cpp

```diff
@@ -977,12 +1174,24 @@ std::string runtime::gpu::CUDAEmitter::include_helpers()
     std::stringstream ss;
 #if defined(CUDA_VERSION) && CUDA_VERSION < 9000
     ss << R"(
+#define WARP_SIZE 32
 #define __ballot_sync(mask, predicate) __ballot(predicate)
 #define __shfl_down_sync(mask, val, delta, width) __shfl_down(val, delta, width)
 #define __shfl_xor_sync(mask, val, laneMask, width) __shfl_xor(val, laneMask, width)
 )";
 #endif
 
+    // add modern type definitions
+    ss << "typedef signed char int8_t;\n";
```

Comment:
These are defined in header files. #include <inttypes> or whatever

Reply:
we can use a .h file for cuda header, than pass to nvrtc.